features/net/network-socket/nsapi_dns.cpp

```diff
@@ -44,6 +45,111 @@ extern "C" int nsapi_dns_add_server(nsapi_addr_t addr)
     return 0;
 }
 
+// IP parsing
+static bool dns_parse_ipv4(const char *host, nsapi_addr_t *addr)
+{
+    int i = 0;
+
+    // Check each digit for [0-9.]
+    for (; host[i]; i++) {
+        if (!(host[i] >= '0' && host[i] <= '9') && host[i] != '.') {
+            return false;
+        }
+    }
+
+    // Ending with '.' garuntees host
+    if (i > 0 && host[i-1] == '.') {
+        return false;
+    }
+
+    // Build up the ip address
+    addr->version = NSAPI_IPv4;
+    i = 0;
+
+    for (int count = 0; count < NSAPI_IPv4_BYTES; count++) {
+        int scanned = sscanf(&host[i], "%hhu", &addr->bytes[count]);
+        if (scanned < 1) {
+            return true;
```

Comment:
Actually this was to handle shortened forms of IP addresses:
http://man7.org/linux/man-pages/man3/inet_aton.3.html#DESCRIPTION

However, looking over the spec again, it looks like the above parser doesn't handle the different formats correctly due to only scanning bytes. Do you think we should just drop the old format and stick to 4 digit ipv4 addresses? ie `4.15.43.83` vs `4.994131`?